File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/supply/dto/SupplyDto.java
package vn.com.kns.portalapi.application.service.supply.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.category.item.dto.ItemDto;
import vn.com.kns.portalapi.application.service.category.unit.dto.UnitDto;
import vn.com.kns.portalapi.application.service.supplier.dto.SupplierDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SupplyDto extends BaseEntityDto {
    private Long price;

    private ItemDto item;
    private Long itemId;

    private SupplierDto supplier;
    private Long supplierId;

    private UnitDto unit;
    private Long unitId;

    private String barcode;
}

File: src/msa/web-service/portalweb/src/app/shared/services/category/supply/dto/supply-dto.ts
import {SupplierDto} from '../../../supplier/dto/supplier-dto';
import {UnitDto} from '../../unit/dto/unit-dto';
import {ItemDto} from '../../item/dto/item-dto';

export class SupplyDto {
  id: number | null;
  price: number = 1;
  barcode: string;

  unitId: number;
  unit: UnitDto = new UnitDto();

  itemId: number;
  item: ItemDto = new ItemDto();

  supplierId: number;
  supplier: SupplierDto = new SupplierDto();
}
